app/routes.py

```diff
@@ -1,2 +1,234 @@
+
+@goals_bp.route("", methods=["GET"], strict_slashes=False)
+def get_goals():
+    order_query = request.args.get("sort")
+    if order_query:
+        if order_query == "asc":
+            goals = Goal.query.order_by(asc(Goal.title))
+        elif order_query == "desc":
+            goals = Goal.query.order_by(desc(Goal.title))
+        else: 
+            raise TypeError("Only asc or desc is accepted here")
+    else:
+        goals = Goal.query.all()
+    goals_response = [goal.to_json() for goal in goals]
+    return jsonify(goals_response), 200
+  
+@goals_bp.route("/<goal_id>", methods=["GET"], strict_slashes=False)
+def get_single_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        return {"goal": goal.to_json()}, 200
+    else:
+        return jsonify(None), 404
+
+@goals_bp.route("", methods=["POST"], strict_slashes=False)
+def create_goal():
+    request_body = request.get_json()
+    if "title" in request_body:
+        new_goal = Goal(title = request_body["title"])
+        db.session.add(new_goal)
+        db.session.commit()
+        return {
+                "goal": new_goal.to_json()
+        }, 201
+    else:
+        return {"details": "Invalid data"}, 400    
+      
+@goals_bp.route("/<goal_id>", methods=["DELETE"], strict_slashes=False)
+def delete_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        db.session.delete(goal)
+        db.session.commit()
+        return {
+              "details": f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"
+        }
+    else:
+        return jsonify(None), 404
+      
+
+@goals_bp.route("/<goal_id>", methods=["PUT"], strict_slashes=False)
+def update_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        form_data = request.get_json()
+        goal.title = form_data["title"]
+        db.session.commit()
+        return {
+                "goal": goal.to_json()
+        }, 200
+    else:
+        return jsonify(None), 404     
+      
+#################### Routes for Goals-Related-Task ####################
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes=False)
+def post_tasks_to_goal(goal_id):
+    request_body = request.get_json()
+    for task_id in request_body["task_ids"]:
+        task = Task.query.get(task_id)
+        task.goal_id = goal_id
+    db.session.commit()
+    return {
+            "id": int(goal_id),
+            "task_ids": request_body["task_ids"]
+    }, 200
+    
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"], strict_slashes=False)
+def get_goals_tasks(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return jsonify(None), 404 
+    else:
+        tasks = Task.query.filter_by(goal_id=goal_id)
+        task_list = [task.to_json() for task in tasks]
+    return {"id":int(goal_id),"title":goal.title,"tasks":task_list},200
+  
+#################### Helper function #################### 
+def validate_data(request_body):
```

Comment:
This function will raise an exception for invalid data, but nothing is catching the error.